### DISCUSSION TITLE
Errors in Installation Test

### DISCUSSION THREAD

**Discussion by michnoffn** [IDAES/idaes-pse#1082] [Q&A]
Hi,

The following attachment is the output of running the test right after completing installation. What am I supposed to do to fix this? Thank you for the help!

![image](https://user-images.githubusercontent.com/116414164/216854413-0ce38959-756a-4298-bdd5-e08443428115.png)

**Comment by andrewlee94** [comment]
@michnoffn We will need some more information to help with debugging that - I have never seen that error message before and I have no idea why it might be showing up on those tests. Specifically - it looks like the issue is in the construction of the model being used for the test rather than the test itself.

First, what operating system and Python version are you using?
Second, what method did you use to install IDAES? The first thing I would do is check the versions of all the dependencies you have installed.
Thirdly, could you provide a full traceback for on of the test failures please? The summary only tells us the final state, not the path the issue took to get there.

**Reply by michnoffn** [reply]
I am currently on Windows 11, the most recent update as of two days ago. I am using Python 3.10.9.

I used the installation guide found https://idaes-pse.readthedocs.io/en/2.0.0b2/tutorials/getting_started/windows.html#windows-installation-guide (not the quickstart). I followed the steps exactly.

Where would I find a full trace back? I'm a little new to programming.

Thank you so much for the help!

**Reply by andrewlee94** [reply]
For the traceback(s), there should be a lot of output that came before the summary of the test failures you showed in the screenshot. Most of that is tracebacks for each of the failures that was encountered.

If you just scroll to the top of the output, it should list the errors first, and you should see red line separators between each failure/error. If you could send us a copy of one of those failures, it would help us see where things are going wrong.

**Reply by michnoffn** [reply]
Is there a report of past tests that show that, or would I have to run the whole test again?

**Reply by andrewlee94** [reply]
There might be a report, but I am not sure where to find it. However, you can run a single test in isolation as well be doing:

`pytest idaes/core/util/tests/test_model_statistics.py -k "test_total_blocks_set"`

The first part tells `pytest` to only look at a single file, and the `-k` tells it to only run tests which match the following name.

**Comment by lbianchi-lbl** [comment]
Hi @michnoffn, thanks for bringing this up.

Could you try to run the following commands in a terminal, and report the results here (either copy-pasting the text, or as a screenshot)?

- A: `pip list`
- B: `pip show idaes-pse pyomo`
- C: `idaes environment-info`

**Reply by michnoffn** [reply]
A:
anyio                              3.6.2
appdirs                            1.4.4
argon2-cffi                        21.3.0
argon2-cffi-bindings               21.2.0
attrs                              22.2.0
backcall                           0.2.0
backports.functools-lru-cache      1.6.4
backports.shutil-get-terminal-size 1.0.0
beautifulsoup4                     4.11.2
bleach                             6.0.0
brotlipy                           0.7.0
bunch                              1.0.1
certifi                            2022.12.7
cffi                               1.15.1
charset-normalizer                 2.1.1
click                              8.1.3
colorama                           0.4.6
comm                               0.1.2
cryptography                       39.0.0
cycler                             0.11.0
debugpy                            1.6.6
decorator                          5.1.1
defusedxml                         0.7.1
entrypoints                        0.4
et-xmlfile                         1.1.0
exceptiongroup                     1.1.0
fastjsonschema                     2.16.2
Flask                              2.2.2
Flask-Cors                         3.0.10
flit_core                          3.8.0
fonttools                          4.38.0
idaes-examples                     1.13.0
idaes-pse                          1.13.0
idna                               3.4
importlib-metadata                 6.0.0
importlib-resources                5.10.2
iniconfig                          2.0.0
ipykernel                          6.21.1
ipython                            7.33.0
ipython-genutils                   0.2.0
ipywidgets                         8.0.4
itsdangerous                       2.1.2
jedi                               0.18.2
Jinja2                             3.1.2
jsonschema                         4.17.3
jupyter                            1.0.0
jupyter_client                     8.0.2
jupyter-console                    6.4.4
jupyter_core                       5.2.0
jupyter-events                     0.6.3
jupyter_server                     2.2.1
jupyter_server_terminals           0.4.4
jupyterlab-pygments                0.2.2
jupyterlab-widgets                 3.0.5
kiwisolver                         1.4.4
lxml                               4.9.1
MarkupSafe                         2.1.2
matplotlib                         3.5.3
matplotlib-inline                  0.1.6
mistune                            2.0.4
mock                               5.0.1
mpmath                             1.2.1
munkres                            1.1.4
nbclassic                          0.5.1
nbclient                           0.7.2
nbconvert                          7.2.9
nbformat                           5.7.3
nest-asyncio                       1.5.6
networkx                           3.0
nose                               1.3.7
notebook                           6.5.2
notebook_shim                      0.2.2
numpy                              1.24.1
openpyxl                           3.0.10
packaging                          23.0
pandas                             1.5.3
pandocfilters                      1.5.0
parso                              0.8.3
pickleshare                        0.7.5
Pillow                             9.3.0
Pint                               0.20.1
pip                                22.3.1
pkgutil_resolve_name               1.3.10
platformdirs                       2.6.2
pluggy                             1.0.0
ply                                3.11
pooch                              1.6.0
prometheus-client                  0.16.0
prompt-toolkit                     3.0.36
psutil                             5.9.4
pycparser                          2.21
Pygments                           2.14.0
Pyomo                              6.4.4
pyOpenSSL                          23.0.0
pyparsing                          3.0.9
pyrsistent                         0.19.3
PySocks                            1.7.1
pytest                             7.2.1
python-dateutil                    2.8.2
python-json-logger                 2.0.4
python-slugify                     8.0.0
pytz                               2022.7.1
PyUtilib                           6.0.0
pywin32                            304
pywinpty                           2.0.10
PyYAML                             6.0
pyzmq                              25.0.0
qtconsole                          5.4.0
QtPy                               2.3.0
rbfopt                             4.2.1
requests                           2.28.2
rfc3339-validator                  0.1.4
rfc3986-validator                  0.1.1
scipy                              1.10.0
Send2Trash                         1.8.0
setuptools                         65.6.3
sip                                4.19.13
six                                1.16.0
sniffio                            1.3.0
soupsieve                          2.3.2.post1
sympy                              1.11.1
terminado                          0.15.0
text-unidecode                     1.3
tinycss2                           1.2.1
tinydb                             4.7.1
tomli                              2.0.1
tornado                            6.2
traitlets                          5.9.0
typing_extensions                  4.4.0
unicodedata2                       15.0.0
Unidecode                          1.3.6
urllib3                            1.26.14
wcwidth                            0.2.6
webencodings                       0.5.1
websocket-client                   1.5.1
Werkzeug                           2.2.2
wheel                              0.37.1
widgetsnbextension                 4.0.5
win-inet-pton                      1.1.0
wincertstore                       0.2
zipp                               3.12.0




B:
Name: idaes-pse
Version: 1.13.0
Summary: IDAES Process Systems Engineering Framework
Home-page: https://idaes.org
Author:
Author-email:
License: BSD
Location: c:\users\nmich\labprograms\miniconda3\envs\idaes\lib\site-packages
Requires: backports.shutil-get-terminal-size, bunch, click, colorama, flask, flask-cors, ipython, jupyter, lxml, matplotlib, nbconvert, nbformat, networkx, numpy, omlt, openpyxl, pandas, pint, psutil, pyomo, pytest, python-slugify, pyyaml, rbfopt, requests, scipy, sympy, tinydb, xlrd
Required-by:
---
Name: Pyomo
Version: 6.4.4
Summary: Pyomo: Python Optimization Modeling Objects
Home-page: http://pyomo.org
Author:
Author-email:
License: BSD
Location: c:\users\nmich\labprograms\miniconda3\envs\idaes\lib\site-packages
Requires: ply
Required-by: idaes-pse, rbfopt




C:
IDAES
    Version: 1.13.0
    Git Hash: None
    Binary Directory: C:\Users\nmich\AppData\Local\idaes\bin
    Data Directory: C:\Users\nmich\AppData\Local\idaes
    Global Config: C:\Users\nmich\AppData\Local\idaes\idaes.conf

Pyomo
    Version: 6.4.4

Python
    Version: 3.10.9 | packaged by conda-forge | (main, Jan 11 2023, 15:15:40) [MSC v.1916 64 bit (AMD64)]
    Executable: C:\Users\nmich\LabPrograms\miniconda3\envs\idaes\python.exe

OS
    Platform: Windows
    Release: 10
    Version: 10.0.22621

Dependencies
    backports.shutil-get-terminal-size: 1.0.0
    bunch: 1.0.1
    click: 8.1.3
    colorama: 0.4.6
    flask: 2.2.2
    flask-cors: 3.0.10
    ipython: 7.33.0
    jupyter: 1.0.0
    lxml: 4.9.1
    matplotlib: 3.5.3
    nbconvert: 7.2.9
    nbformat: 5.7.3
    networkx: 3.0
    numpy: 1.24.1
    omlt: Not Installed
    openpyxl: 3.0.10
    pandas: 1.5.3
    pint: 0.20.1
    psutil: 5.9.4
    pytest: 7.2.1
    python-slugify: 8.0.0
    pyyaml: 6.0
    rbfopt: 4.2.1
    requests: 2.28.2
    scipy: 1.10.0
    sympy: 1.11.1
    tinydb: 4.7.1
    xlrd: Not Installed

Extras
    seaborn: Not Installed

Solvers
    bonmin: 1.8.8.0
    cbc: 2.10.4.0
    couenne: 0.5.8.0
    dot_sens: 1.0.0.0
    ipopt: 3.13.2.0
    ipopt_l1: 3.13.3.0
    ipopt_sens: 3.13.2.0
    k_aug: 0.1.0.0

**Reply by lbianchi-lbl** [reply]
Thanks for running the checks. The first thing that jumps to my eye is that the installed version of `idaes-pse` is 1.13.0. I don't know when the installation was performed, but IDAES v1 is now considered unsupported.

To uninstall IDAES v1 and reinstall the best available version of IDAES v2 (2.0.0b2, with the stable 2.0.0 coming soon later this month), run the following commands in the same Conda environment:

```sh
pip uninstall idaes-pse --yes  && pip install idaes-pse --pre
idaes get-extensions --verbose --extra petsc
```

Then, you can use the same commands as above (`pip show idaes-pse`, etc) to verify that the versions of IDAES and Pyomo have been updated.

**Reply by michnoffn** [reply]
That's really weird, the installation was performed two days ago! I reinstalled it, and it shows the correct version this time:
(idaes) C:\Users\nmich\LabPrograms>pip show idaes-pse
Name: idaes-pse
Version: 2.0.0b2
Summary: IDAES Process Systems Engineering Framework
Home-page: https://idaes.org
Author:
Author-email:
License: BSD
Location: c:\users\nmich\labprograms\miniconda3\envs\idaes\lib\site-packages
Requires: backports.shutil-get-terminal-size, bunch, click, colorama, distro, flask, flask-cors, ipython, jupyter, lxml, matplotlib, nbconvert, nbformat, networkx, numpy, omlt, openpyxl, pandas, pint, psutil, pyomo, pytest, python-slugify, pyyaml, rbfopt, requests, scipy, sympy, tinydb, xlrd
Required-by:

I will re run the test and get back to you.

**Reply by lbianchi-lbl** [reply]
OK, it looks like the correct version should be installed now.

Just to be clear, the reason why I asked when the (original) installation had been performed is that, since mid-December last year, I would expect one of these versions to be installed by default with `pip install idaes-pse`, depending on whether the `--pre` flag is used or not:

- If the `--pre` flag is given (see the [release notes](https://github.com/IDAES/idaes-pse/releases/tag/2.0.0b2)): 2.0.0b2
- If the `--pre` flag is not given: 1.13.2

Since the default behavior of `pip install` is without the `--pre` flag, in 1.13.2 we added a warning about the v1 -> v2 transition that's displayed whenever `idaes` or one of its submodules is imported. However, from your original `pip show`, the reported version seems to be 1.13.0, which being lower than 1.13.2, should not have been installed by default :thinking:

**Reply by michnoffn** [reply]
Here is the updated failure report, is this acceptable or should I fix it?

![image](https://user-images.githubusercontent.com/116414164/217659313-3067d226-65aa-48d5-88e2-c3b013d9056b.png)

**Reply by lbianchi-lbl** [reply]
After discussing with @dangunter, our best guess is that this failure might be due to the `C:\Users\...\.idaes\_scratch\...` directory not being cleaned up properly during or after the tests. Therefore, it should be safe to disregard the failures.
